File: android/app/src/main/java/info/bagen/rust/plaoc/microService/core/BootstrapContext.kt
package info.bagen.rust.plaoc.microService.core

import info.bagen.rust.plaoc.microService.helper.Mmid
import info.bagen.rust.plaoc.microService.ipc.Ipc


interface BootstrapContext {
    val dns: DnsMicroModule
}

interface DnsMicroModule {
    /**
     * 动态安装应用
     */
    fun install(mm: MicroModule)

    /**
     * 动态卸载应用
     */
    fun uninstall(mm: MicroModule)

    /**
     * 与其它应用建立连接
     */
    suspend fun connect(mmid: Mmid): Ipc
}

File: ios/BFExplorer/Plaoc/bfs/microModule/core/BootstrapContext.swift
//
//  BootstrapContext.swift
//  BFExplorer
//
//  Created by ui08 on 2023/3/10.
//

import Foundation
import Vapor

protocol DnsMicroModule {
    /**
     * 动态安装应用
     */
    func install(mm: MicroModule)
    /**
     * 动态卸载应用
     */
    func uninstall(mm: MicroModule)
    /**
     * 与其它应用建立连接
     */
    func connect(mmid: Mmid, reason: Request?) async -> ConnectResult
}

protocol BootStrapContext {
    var dns: DnsMicroModule { get }
}
